doc/development/deprecations.rst

```diff
@@ -0,0 +1,83 @@
+.. _deprecations:
+
+Deprecations
+============
+
+Pending Deprecations
+--------------------
+
+* In-place inversion is now deprecated. This includes ``op.inv()`` and ``op.inverse=value``. Please
+  use ``qml.adjoint`` or ``qml.pow`` instead. 
+
+  - Still accessible in v0.27 and v0.28
+  - Removed in v0.29
+
+  Don't use:
+
+  >>> v1 = qml.PauliX(0).inv()
+  >>> v2 = qml.PauliX(0)
+  >>> v2.inverse = True
+
+  Instead use:
+
+  >>> qml.adjoint(qml.PauliX(0))
+  Adjoint(PauliX(wires=[0]))
+  >>> qml.pow(qml.PauliX(0), -1)
+  PauliX(wires=[0])**-1
+  >>> qml.pow(qml.PauliX(0), -1, lazy=False)
+  PauliX(wires=[0])
+  >>> qml.PauliX(0) ** -1
+  PauliX(wires=[0])**-1
+
+* ``QueuingContext`` is renamed ``QueuingManager``. 
+
+  - Still accessible via ``QueuingContext`` in v0.27
+  - Only accessible via ``QueuingManager`` in v0.28
+
+* ``QueuingManager.safe_update_info`` and ``AnnotateQueue.safe_update_info`` are removed.
+
+  - Still accessible in v0.27
+  - Removed in v0.28
+
+* ``qml.tape.stop_recording`` and ``QuantumTape.stop_recording`` are moved to ``qml.QueuingManager.stop_recording``
+
+  - Still accessible in v0.27
+  - Removed in v0.28
+
+* ``qml.tape.get_active_tape`` is deprecated. Please use ``qml.QueuingManager.active_context()`` instead.
+
+  - Still accessible in v0.27
+  - Removed in v0.28
+
+* ``qml.ExpvalCost`` has been deprecated, and usage will now raise a warning.
+  
+  - Deprecated in v0.24
+  - Will be removed in v0.28
+
+  Instead, it is recommended to simply
+  pass Hamiltonians to the ``qml.expval`` function inside QNodes:
+
+  .. code-block:: python
+
+    @qml.qnode(dev)
+    def ansatz(params):
+        some_qfunc(params)
+        return qml.expval(Hamiltonian)
+
+
+Completed Deprecation Cycles
```

Comment:
I'm curious the motivation for retaining this here -- this could potentially become a _very_ big file!

Reply:
I think it wouldn't hurt. I mean.. github will keep the historical record for us, but the size of this nearly plaintext file doesn't scare me too much compared to the size of other problems we face in PennyLane 😄

Reply:
If it starts to get too big, we can get rid of old ones, but I don't think it hurts at this point.